GameProject/Assets/Turn based assets/Scripts/MouseSelection.cs

```diff
@@ -0,0 +1,63 @@
+using System;
+using UnityEngine;
+
+namespace Turn_based_assets.Scripts
+{
+    public class MouseSelection : MonoBehaviour
+    {
+        private ISelection _selection;
+        [SerializeField] private PlayerController player;
+        public Vector3 rawGridPoint;
+        public float distanceY;
+        public float offset;
+        public float gridSize;
+        public Plane _plane;
+        private Vector3 distanceFromCamera;
+        
+
+        private void Start()
+        {
+            var position = Camera.main.transform.position;
+            distanceFromCamera = new Vector3(position.x, position.y - distanceY,position.z);
+            _plane = new Plane(Vector3.up, distanceFromCamera);
+        }
+
+        private void Update()
+        {
+            Ray ray = Camera.main.ScreenPointToRay(Input.mousePosition);
+
+            float gridSpace = 0;
+
+            if (_plane.Raycast(ray, out gridSpace))
+            {
+                rawGridPoint = ray.GetPoint(gridSpace);
```

Comment:
This seems like an odd reuse of the `rawGridPoint` field - with the code commented out, it seems like the field doesn't need to be scoped to the behavior anymore and could be placed in in scope of the method instead.